middle_end/flambda2/to_cmm/to_cmm_shared.ml

```diff
@@ -389,12 +409,16 @@ let make_update env res dbg ({ kind; stride } : Update_kind.t) ~symbol var
         match kind with
         | Pointer -> Word_val
         | Immediate -> Word_int
+        | Naked_int8 ->
+          if stride = Arch.size_addr then Word_int else Byte_signed
```

Comment:
Considering the definitions above, the `else` branch of this condition is dead code. Since that code will then not be used and tested for a while, i'd rather see it fail at compile time, rather than have dead code that may work in the future if the correct changes are made (it will be simpler to write the correct code at that point).